packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -334,54 +341,33 @@ class FlutterPlugin implements Plugin<Project> {
                     provided project.files(flutterJar)
                 }
             } else {
+                assert baseJar["debug"].isFile()
+                assert baseJar["profile"].isFile()
+                assert baseJar["release"].isFile()
+
                 if (project.getConfigurations().findByName("debugCompileOnly")) {
-                    debugCompileOnly project.files(debugFlutterJar)
-                    profileCompileOnly project.files(profileFlutterJar)
-                    releaseCompileOnly project.files(releaseFlutterJar)
+                    debugCompileOnly project.files(baseJar["debug"])
```

Comment:
nit, subjective/optional: I think having the 20 different variables per Jar was more verbose, but ultimately safer than this. If there's a typo in either the setting or getting of any of these string keys this will end up being null, and the reason why will be hard to debug because the language won't complain about a mismatched key. With named variables it'll at least fail harder and faster if there's a typo in the variable in any of these places. You could guard against it some and still keep the map by keeping these constants, the assert you added will help catch it too.

Reply:
I added a `TODO` to figure out what to do with this code as part of the refactoring. Agreed, strings are less safe, but I'd like to eventually remove this code, or change the way it's structured: https://github.com/flutter/flutter/issues/33064